src/plugins/data/public/search/search_interceptor.ts

```diff
@@ -172,34 +173,95 @@ export class SearchInterceptor {
 
     const combinedSignal = getCombinedSignal(signals);
     const cleanup = () => {
-      this.timeoutSubscriptions.remove(subscription);
+      subscription.unsubscribe();
     };
 
     combinedSignal.addEventListener('abort', cleanup);
 
     return {
       combinedSignal,
+      timeoutSignal,
       cleanup,
     };
   }
 
-  // Right now we are debouncing but we will hook this up with background sessions to show only one
-  // error notification per session.
-  protected showTimeoutError = debounce(
-    (e: Error) => {
-      this.deps.toasts.addError(e, {
+  /**
+   * Right now we are debouncing but we will hook this up with background sessions to show only one
+   * error notification per session.
+   * @internal
+   */
+  private showTimeoutError = debounce(
```

Comment:
I think this doesn't work us intended and this is the reason of https://github.com/elastic/kibana/issues/77579

I guess instead of `debounce` we need `throttle` here

Reply:
It's not the reason for the bug, the reason was that we didn't unsubscribe properly all together here https://github.com/elastic/kibana/pull/77788/files?file-filters%5B%5D=.js&file-filters%5B%5D=.ts&file-filters%5B%5D=.tsx#diff-75f11add5ff5d57a12822fe0c5a52536R176

But I agree that throttle sounds like a better match.